dashboard/app/controllers/projects_controller.rb

```diff
@@ -54,10 +54,7 @@ def angular
   end
 
   def load
-    if STANDALONE_PROJECTS[params[:key]][:login_required]
-      authenticate_user!
-    end
-    return if redirect_unless_account_type_requirements_met
+    authorize! :load_project, params[:key]
```

Comment:
But wait... how does this interact with a rescue case in the methods in question?